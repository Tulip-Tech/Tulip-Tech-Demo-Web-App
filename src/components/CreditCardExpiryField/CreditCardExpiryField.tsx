import React from 'react';

import TextField from '../TextField/TextField';

type Props = {
  onChange?: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  onBlur?: React.FocusEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  error?: string;
  value?: string;
};

const CreditCardExpiryField: React.FC<Props> = ({ value, onChange, error, ...props }: Props) => {
  const formatExpirationDate: React.ChangeEventHandler<HTMLInputElement> = (e) => {
    let clearValue = e.target.value.replace(/\D+/g, '');

    if (clearValue.length === 1) {
      if (Number(clearValue) > 1) {
        clearValue = '0' + clearValue;
      }
    } else if (clearValue.length === 2) {
      if (Number(clearValue) > 12) {
        clearValue = '12';
      }
    }

    if (clearValue.length >= 3) {
      e.target.value = `${clearValue.slice(0, 2)}/${clearValue.slice(2, 4)}`;
    } else {
      e.target.value = clearValue;
    }
    if (onChange) {
      onChange(e);
    }
  };
  return (
    <TextField
      label={`Expiry date`}
      {...props}
      name="cardExpiry"
      value={value}
      error={!!error}
      helperText={error ? error : null}
      onChange={formatExpirationDate}
      type="text"
      pattern="[0-1][0-9]/[0-9]{2}"
      placeholder="MM/YY"
      required
    />
  );
};

export default CreditCardExpiryField;
