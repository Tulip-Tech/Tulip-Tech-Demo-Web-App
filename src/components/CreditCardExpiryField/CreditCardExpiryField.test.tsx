import React from 'react';
import { render } from '@testing-library/react';

import CreditCardExpiryField from './CreditCardExpiryField';

describe('<CreditCardExpiryField>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<CreditCardExpiryField value="" onChange={vi.fn()} />);

    expect(container).toMatchSnapshot();
  });
});
