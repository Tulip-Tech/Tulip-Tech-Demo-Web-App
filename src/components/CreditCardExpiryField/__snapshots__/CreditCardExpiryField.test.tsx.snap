// Vitest Snapshot v1

exports[`<CreditCardExpiryField> > renders and matches snapshot 1`] = `
<div>
  <div
    class="_textField_e16c1b"
  >
    <label
      class="_label_e16c1b"
      for="text-field_1235_cardExpiry"
    >
      Expiry date
    </label>
    <div
      class="_container_e16c1b"
    >
      <input
        class="_input_e16c1b"
        id="text-field_1235_cardExpiry"
        name="cardExpiry"
        pattern="[0-1][0-9]/[0-9]{2}"
        placeholder="MM/YY"
        required=""
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
